fdbserver/Resolver.actor.cpp

```diff
@@ -310,21 +312,22 @@ ACTOR Future<Void> resolveBatch(Reference<Resolver> self, ResolveTransactionBatc
 		auto& stateTransactions = stateTransactionsPair.second;
 		int64_t stateMutations = 0;
 		int64_t stateBytes = 0;
-		LogPushData toCommit(self->logSystem); // For accumulating private mutations
+		std::unique_ptr<LogPushData> toCommit(nullptr); // For accumulating private mutations
+		bool isLocked = false;
+		if (SERVER_KNOBS->PROXY_USE_RESOLVER_PRIVATE_MUTATIONS) {
+			auto lockedKey = self->txnStateStore->readValue(databaseLockedKey).get();
+			isLocked = lockedKey.present() && lockedKey.get().size();
+			toCommit.reset(new LogPushData(self->logSystem));
+		}
 		ResolverData resolverData(self->dbgid,
```

Comment:
Can we put this inside the above block too? Seems this is also VV related only.

Reply:
Good idea. Done.